### PR TITLE
KOGITO-5280 Setup downstream profiles for kogito checks

### PR DESCRIPTION
**Pull request by radtriste**
### JIRA

https://issues.redhat.com/browse/KOGITO-5280

### Referenced pull requests

https://github.com/kiegroup/kogito-apps/pull/1033
https://github.com/kiegroup/kogito-examples/pull/897
https://github.com/kiegroup/kogito-pipelines/pull/271
https://github.com/kiegroup/kogito-apps/pull/1049
https://github.com/kiegroup/optaplanner-quickstarts/pull/191

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

**Review by rsynek — APPROVED** [review]
LGTM, thank you!

Should we also add these profiles to the BuildChain-based downstream build? On one hand, it's currently disabled, thus we don't need this optimization there at all. But on the other hand, the CI should be kept in sync, ideally being interchangeable.
This is an example of how two different downstream build systems increase the maintenance cost, unfortunately.

**Comment by radtriste** [comment]
@rsynek What do you mean exactly ? what do you call the `BuildChain-based downstream build` ?

**Comment by rsynek** [comment]
@radtriste I meant this one:
https://github.com/kiegroup/optaplanner/blob/main/.github/workflows/full-downstream.yml
Shall I rather call it the GHA-based full-downstream build, as both downstream builds use the build chain tool nowadays?

**Comment by radtriste** [comment]
> @radtriste I meant this one: https://github.com/kiegroup/optaplanner/blob/main/.github/workflows/full-downstream.yml Shall I rather call it the GHA-based full-downstream build, as both downstream builds use the build chain tool nowadays?

oh yeah sure. gonna update it

**Comment by radtriste** [comment]
jenkins rerun kogito-apps tests

**Comment by radtriste** [comment]
jenkins rerun kogito-apps tests

**Comment by radtriste** [comment]
Examples => ok
Apps => created https://github.com/kiegroup/kogito-apps/pull/1049 to correct the list of modules

File: .ci/jenkins/dsl/jobs.groovy [hunk 1]
```diff
@@ -96,6 +102,9 @@ void setupMultijobPrDefaultChecks() {
 }
 
 void setupMultijobPrNativeChecks() {
+    def multijobConfig = getMultijobPRConfig()
+    multijobConfig.jobs.find { it.id == 'kogito-apps' }.env.KOGITO_APPS_BUILD_MVN_OPTS = '-Poptaplanner-downstream,native -am -amd'
+    multijobConfig.jobs.find { it.id == 'kogito-examples' }.env.KOGITO_EXAMPLES_BUILD_MVN_OPTS = '-Poptaplanner-downstream-native -am -amd'
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:107]:
```suggestion
    multijobConfig.jobs.find { it.id == 'kogito-apps' }.env.KOGITO_APPS_BUILD_MVN_OPTS = '-Poptaplanner-downstream,native'
    multijobConfig.jobs.find { it.id == 'kogito-examples' }.env.KOGITO_EXAMPLES_BUILD_MVN_OPTS = '-Poptaplanner-downstream-native'
```

File: .ci/jenkins/dsl/jobs.groovy [hunk 2]
```diff
@@ -27,9 +27,15 @@ Map getMultijobPRConfig() {
             ], [
                 id: 'kogito-apps',
                 repository: 'kogito-apps',
+                env : [
+                    KOGITO_APPS_BUILD_MVN_OPTS: '-Poptaplanner-downstream -am -amd'
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:31]:
```suggestion
                    KOGITO_APPS_BUILD_MVN_OPTS: '-Poptaplanner-downstream'
```

File: .ci/jenkins/dsl/jobs.groovy [hunk 3]
```diff
@@ -27,9 +27,15 @@ Map getMultijobPRConfig() {
...
+                ]
             ], [
                 id: 'kogito-examples',
-                repository: 'kogito-examples'
+                repository: 'kogito-examples',
+                env : [
+                    KOGITO_EXAMPLES_BUILD_MVN_OPTS: '-Poptaplanner-downstream -am -amd'
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:37]:
```suggestion
                    KOGITO_EXAMPLES_BUILD_MVN_OPTS: '-Poptaplanner-downstream'
```

**Comment by radtriste** [comment]
Examples:
```
11:25:46  [INFO] ------------------------------------------------------------------------
11:25:46  [INFO] Reactor Summary for Kogito Example 2.0.0-SNAPSHOT:
11:25:46  [INFO] 
11:25:46  [INFO] Kogito Example ..................................... SUCCESS [  1.664 s]
11:25:46  [INFO] Kogito Example :: Process and Quarkus .............. SUCCESS [01:25 min]
11:25:46  [INFO] Kogito Example :: Process and Spring Boot .......... SUCCESS [ 30.520 s]
11:25:46  [INFO] Kogito Example :: Process, OptaPlanner and Quarkus . SUCCESS [01:13 min]
11:25:46  [INFO] Kogito Example :: Process, OptaPlanner and Spring Boot SUCCESS [ 23.572 s]
11:25:46  [INFO] ------------------------------------------------------------------------
11:25:46  [INFO] BUILD SUCCESS
11:25:46  [INFO] ------------------------------------------------------------------------
11:25:46  [INFO] Total time:  03:36 min
11:25:46  [INFO] Finished at: 2021-10-05T05:25:44-04:00
11:25:46  [INFO] ------------------------------------------------------------------------
```

**Comment by radtriste** [comment]
jenkins rerun kogito-apps tests

**Comment by radtriste** [comment]
jenkins rerun kogito-apps tests

**Pull request merged**